datafusion/expr/src/utils.rs

```diff
@@ -45,6 +45,22 @@ pub fn exprlist_to_columns(expr: &[Expr], accum: &mut HashSet<Column>) -> Result
     Ok(())
 }
 
+/// Count the number of distinct exprs in a list of group by expressions. If the
+/// first element is a `GroupingSet` expression then it must be the only expr.
+pub fn grouping_set_expr_count(group_expr: &[Expr]) -> Result<usize> {
+    if let Some(Expr::GroupingSet(grouping_set)) = group_expr.first() {
+        if group_expr.len() > 1 {
+            return Err(DataFusionError::Plan(
+                "Invalid group by expressions, GroupingSet must be the only expression"
+                    .to_string(),
+            ));
+        }
+        Ok(grouping_set.distinct_expr().len())
+    } else {
+        Ok(group_expr.len())
```

Comment:
Do we need to do other validation? Should I be able to `group by *`?

Reply:
I expect that we could add extra validation here over time. I would need to research what is allowable.

Reply:
FWIW the sqlparser won't allow `group by *`

```
❯ select count(*) from foo group by *;  🤔 Invalid statement: sql parser error: Expected an expression:, found: *
```